src/core/contrib.js

```diff
@@ -10,107 +10,90 @@ import { pub } from "core/pubsubhub";
 export const name = "core/contrib";
 
 function prop(prop) {
-  return function(o) {
-    return o[prop];
-  };
+  return o => o[prop];
 }
 
 function findUsers(...thingsToFind) {
-  var users = {};
-  thingsToFind.forEach(function(things) {
-    things.forEach(function(thing) {
+  const users = new Set();
+  for (const things of thingsToFind) {
```

Comment:
Although this work better than existing code, I wonder if we can .map() the URLs out of things.users and .reduce() into users set? Declaritve loops are easier to rationalize about and work with (for me, at least 😊)

Reply:
I'm not very familiar with `.reduce` but we probably can do:

```js
const urls = thingsToFind
    .map(things => things.filter(thing => thing.user).map(thing => thing.user.url))
    .reduce((x, y) => x.concat(y));
return [...new Set(urls)];
```

Would this be declarative enough?

Reply:
The nesting is still a bit hard to read (in which case the for loop is more clear)... I’ll think about it and propose something tomorrow (😴 2am here) but otherwise, I’m ok for this to land